tornettools/parse_tgen.py

```diff
@@ -43,10 +43,16 @@ def extract_tgen_plot_data(args):
     # parse performance stats only after the network has reached steady state
     startts, stopts = args.converge_time, -1 if args.run_time < 0 else args.converge_time + args.run_time
 
+    if 'data' in data:
+        for name in data['data']:
+            for stream in data['data'][name]['tgen']['streams'].values():
+                tgen_stream_fill_bytes_from_progress(stream)
```

Comment:
Won't this result in a lot of overhead for large experiments that have a lot of progress data? It seems heavy-handed to copy all of the progress data when you only need the 4MB mark.

Why not copy just the 4 MB time?

Reply:
I was shooting for a more general solution of fixing the problem of "maybe the bytes you're looking for is in 'payload_bytes_recv', or maybe it's in 'payload_progress_recv'.

I don't think adding these data points is terribly expensive, but I also wasn't crazy about mutating the streams. Changed it to fall back to payload_progress_recv on the fly instead. I avoided this initially because I was thinking that working around potential floating point decimal rounding issues would make the code a bit complicated. Python rounds when converting to strings anyway though, basically sweeping this issue under the rug https://docs.python.org/3/tutorial/floatingpoint.html .